src/main/java/com/google/api/codegen/config/DiscoveryField.java

```diff
@@ -61,25 +101,44 @@ private DiscoveryField(Schema schema, DiscoApiModel apiModel) {
   }
 
   /** Create a FieldModel object from a non-null Schema object. */
-  public static DiscoveryField create(Schema schema, DiscoApiModel rootApiModel) {
-    Preconditions.checkNotNull(schema);
-    Preconditions.checkNotNull(rootApiModel);
-    return new DiscoveryField(schema, rootApiModel);
+  public static synchronized DiscoveryField create(Schema schema, DiscoApiModel rootApiModel) {
```

Comment:
`synchronized` because the method performs operations on a global `HashMap`. I can't predict how other people are going to use this class, so they could theoretically use multiple threads? I can take this out if it's overkill?